src/net/http.js

```diff
@@ -414,7 +438,25 @@ Object.assign(pc, function () {
         },
 
         _onError: function (method, url, options, xhr) {
-            options.callback(xhr.status, null);
+            if (options.retrying) {
+                return;
+            }
+
+            // retry if necessary
+            if (options.retryable && options.retries < options.maxRetries) {
+                options.retries++;
+                options.retrying = true; // used to stop retrying when both onError and xhr.onerror are called
+                var retryDelay = pc.math.clamp(Math.pow(2, options.retries) * Http.retryDelay, 0, options.maxRetryDelay || 5000);
+                console.log(method + ': ' + url + ' - Error ' + xhr.status + '. Retrying in ' + retryDelay + ' ms');
```

Comment:
Wrap this in #DEBUG?

Reply:
So the reason why I haven't wrapped this is because when the games run then if they capture application logs I was hoping we'd see whether requests are in fact retried and how often. It does rely on the assumption that logs are captured which I'm not sure it actually happens.